privacyidea/lib/policy.py

```diff
@@ -761,9 +761,9 @@ def ui_get_rights(self, scope, realm, username, client=None):
             # We do not have any policies in this scope, so we return all
             # possible actions in this scope.
             log.debug("No policies defined, so we set all rights.")
-            static_rights = get_static_policy_definitions(scope).keys()
-            enroll_rights = get_dynamic_policy_definitions(scope).keys()
-            rights = static_rights + enroll_rights
+            tmp_rights = get_static_policy_definitions(scope)
+            tmp_rights.update(get_dynamic_policy_definitions(scope))
+            rights = list(tmp_rights)
         # reduce the list
         rights = list(set(rights))
```

Comment:
We can drop this line, since the dict ``tmp_rights`` contains unique keys. Using ``list(set())`` was used to make the entris in the list unique.

Reply:
correct

Reply:
What about the block before? Could we get duplicate entries from the
current policies?

Reply:
So it doesn't really matter if some 'rights' occur multiple times in the list?
I meant the block in line 746:757 where we create the list of 'rights' from the current policies.

Reply:
It *should*TM not.
The rights are read in the [auth API](https://github.com/privacyidea/privacyidea/blob/master/privacyidea/api/auth.py#L273) and [returned by the /auth endpoint](https://github.com/privacyidea/privacyidea/blob/master/privacyidea/api/auth.py#L305).

The the UI uses the [checkRight](https://github.com/privacyidea/privacyidea/blob/master/privacyidea/static/components/login/factories/auth.js#L54)-function to determine, if a user e.g. is allowed to set the Token PIN, so that we can display the "set PIN" button or not.

However, it would also probably not hurt to avoid duplicate entries.

Reply:
changed the code so that we have only unique elements in the `rights` list at the end of the function.